### DISCUSSION TITLE
Length of a JSON Array

### DISCUSSION THREAD

**Discussion by retropiepro** [grafana/grafana-json-datasource#113] [Q&A]
Hello, 

Thank you for this fantastic plugin!!!

I'm wondering if it's possible to display a count of a JSON array?

I'm working with this JSON response, and I would like to visualize the count for the following JSON Path query `$.job[?(@.status=='FINISHED')]`

`{
    "hits": 60,
    "job": [
        {
            "jobId": "VX-1",
            "user": "lz0DSJbVIYCoKrsa8cjZxIvwN",
            "started": "2021-04-05T18:23:49.656+0000",
            "status": "FINISHED",
            "type": "PLACEHOLDER_IMPORT",
            "priority": "MEDIUM"
        },
        {
            "jobId": "VX-2",
            "user": "oNNPGlym7r35jWFI8CJmE6RZy",
            "started": "2021-04-05T18:40:48.303+0000",
            "status": "FAILED_TOTAL",
            "type": "PLACEHOLDER_IMPORT",
            "priority": "MEDIUM"
        }
   ]
}
`

Thanks!

**Comment by marcusolsson** [comment]
Thank you for the kind words! ❤️ 

So far I've limited the scope of the plugin to only query data. It sounds like you'd like to compute a new value from the data returned by the data source, which is currently not supported.

I've been asked about potentially supporting JSONata, which seems to have support for this, using the [$count function](https://docs.jsonata.org/array-functions#count). I've created a feature request for this here: https://github.com/marcusolsson/grafana-json-datasource/issues/114
